tests/unit/events/queue_tests.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+import json
+import unittest
+
+from baseplate.events import (
+    Event,
+    EventError,
+    EventQueue,
+    EventQueueFullError,
+    EventTooLargeError,
+    MAX_EVENT_SIZE,
+)
+from baseplate.message_queue import MessageQueue, TimedOutError
+
+from ... import mock
+
+
+class EventTests(unittest.TestCase):
+    @mock.patch("uuid.uuid4")
+    @mock.patch("time.time")
+    def test_minimal_constructor(self, time, uuid):
+        time.return_value = 333
+        uuid.return_value = "1-2-3-4"
+
+        event = Event("topic", "type")
+
+        self.assertEqual(event.topic, "topic")
+        self.assertEqual(event.event_type, "type")
+        self.assertEqual(event.timestamp, 333000)
+        self.assertEqual(event.id, "1-2-3-4")
+
+    def test_timestamp_specified(self):
+        timestamp = datetime.datetime(2016, 2, 23, 0, 3, 17)
+        event = Event("topic", "type", timestamp=timestamp)
+        self.assertEqual(event.timestamp, 1456185797000)
+
```

Comment:
Yeah, it would. I just don't know a good way to do that without either a) requiring `pytz` to run the test suite, b) making my own timezone class, c) only testing that on py3.2+ which has a timezone in the stdlib. Thoughts?